qp/ensemble.py

```diff
@@ -0,0 +1,519 @@
+import numpy as np
+from pathos.multiprocessing import ProcessingPool as Pool
+import psutil
+import os
+# import sqlalchemy
+import scipy.interpolate as spi
+import matplotlib.pyplot as plt
+
+import qp
+
+class Ensemble(object):
+
+    def __init__(self, N, pdfs=None, truth=None, quantiles=None, histogram=None, gridded=None, samples=None, scheme='linear', vb=True, procs=None):# where='ensemble.db', procs=None):#
+        """
+        Creates an object comprised of many qp.PDF objects to efficiently
+        perform operations on all of them
+
+        Parameters
+        ----------
+        N: int
+            number of pdfs in the ensemble
+        truth: list of scipy.stats.rv_continuous objects or qp.composite objects
+            , optional
+            List of length (npdfs) containing the continuous, parametric forms of the PDFs
+        quantiles: tuple of ndarrays, optional
+            Pair of arrays of lengths (nquants) and (npdfs, nquants) containing
+            shared CDF values and quantiles for each pdf
+        histogram: tuple of ndarrays, optional
+            Pair of arrays of lengths (nbins+1) and (npdfs, nbins) containing
+            shared endpoints of bins and values in bins for each pdf
+        gridded: tuple of ndarrays, optional
+            Pair of arrays of lengths (npoints) and (npdfs, npoints) containing
+            shared points at which pdfs are evaluated and the values of each
+            pdf at those points
+        samples: ndarray, optional
+            Array of size (npdfs, nsamples) containing sampled values
+        scheme: string, optional
+            name of interpolation scheme to use.
+        vb: boolean, optional
+            report on progress
+        where: string
+            path to file corresponding to Ensemble, presumed to not yet exist
+        procs: int, optional
+            limit the number of processors used, otherwise use all available
+        """
```

Comment:
There are now notes under `qp.Ensemble.__init__()` explaining what the ensemble is and what it will hopefully develop into in the future, as #48 and other issues are addressed.